apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/UrlSignerTest.cs

```diff
@@ -0,0 +1,123 @@
+                GoogleCredential.GetApplicationDefaultAsync().Result.UnderlyingCredential as ServiceAccountCredential);
+
+            Assert.Throws<ArgumentNullException>(() => signer.Sign(null, "objectName", TimeSpan.FromDays(1), new HttpRequestMessage()));
+            Assert.Throws<ArgumentException>(() => signer.Sign("BUCKETNAME", "objectName", TimeSpan.FromDays(1), new HttpRequestMessage()));
+
+            Assert.Throws<ArgumentNullException>(() => signer.Sign(null, "objectName", DateTimeOffset.UtcNow, new HttpRequestMessage()));
+            Assert.Throws<ArgumentException>(() => signer.Sign("BUCKETNAME", "objectName", DateTimeOffset.UtcNow, new HttpRequestMessage()));
+
+            var emptyHeaders = new Dictionary<string, IEnumerable<string>>();
+            Assert.Throws<ArgumentNullException>(() => signer.Sign(null, "objectName", TimeSpan.FromDays(1), HttpMethod.Get, emptyHeaders, emptyHeaders));
+            Assert.Throws<ArgumentException>(() => signer.Sign("BUCKETNAME", "objectName", TimeSpan.FromDays(1), HttpMethod.Get, emptyHeaders, emptyHeaders));
+
+            Assert.Throws<ArgumentNullException>(() => signer.Sign(null, "objectName", DateTimeOffset.UtcNow, HttpMethod.Get, emptyHeaders, emptyHeaders));
+            Assert.Throws<ArgumentException>(() => signer.Sign("BUCKETNAME", "objectName", DateTimeOffset.UtcNow, HttpMethod.Get, emptyHeaders, emptyHeaders));
+
+            // Make sure exceptions are not thrown for things which may be null or uppercase.
+            signer.Sign("bucketname", null, TimeSpan.FromDays(1), null);
+            signer.Sign("bucketname", null, null, null, null, null);
+            signer.Sign("bucketname", "OBJECTNAME", null, null, null, null);
+        }
+
+        [Fact]
+        public void DefaultHttpMethodIsGet()
+        {
+            var signer = UrlSigner.FromServiceAccountCredential(
+                GoogleCredential.GetApplicationDefaultAsync().Result.UnderlyingCredential as ServiceAccountCredential);
+
+            var bucketName = "bucket-name";
+            var objectName = "object-name";
+            var expiration = DateTimeOffset.UtcNow + TimeSpan.FromDays(1);
+            var url1 = signer.Sign(bucketName, objectName, expiration, requestMethod: null);
+            var url2 = signer.Sign(bucketName, objectName, expiration, requestMethod: HttpMethod.Get);
+            Assert.Equal(url1, url2);
+        }
+
+        [Fact]
+        public void EncryptionKeyAndHashAreIgnored()
+        {
+            var signer = UrlSigner.FromServiceAccountCredential(
+                GoogleCredential.GetApplicationDefaultAsync().Result.UnderlyingCredential as ServiceAccountCredential);
+
+            var bucketName = "bucket-name";
+            var objectName = "object-name";
+            var expiration = DateTimeOffset.UtcNow + TimeSpan.FromDays(1);
+            var url1 = signer.Sign(bucketName, objectName, expiration, requestHeaders: new Dictionary<string, IEnumerable<string>> {
```

Comment:
I'm pretty sure the unit test project has internal access to the EncryptionKey constants...